allennlp/data/tokenizers/pretrained_transformer_tokenizer.py

```diff
@@ -22,36 +22,101 @@ class PretrainedTransformerTokenizer(Tokenizer):
+        - 'only_first': Only truncate the first sequence
+        - 'only_second': Only truncate the second sequence
+        - 'do_not_truncate': Do not truncate (raise an error if the input sequence is longer than max_length)
+
+    Argument descriptions are from
+    https://github.com/huggingface/transformers/blob/155c782a2ccd103cf63ad48a2becd7c76a7d2115/transformers/tokenization_utils.py#L691
     """
 
     def __init__(
-        self, model_name: str, start_tokens: List[str] = None, end_tokens: List[str] = None
+        self,
+        model_name: str,
+        add_special_tokens: bool = True,
+        max_length: int = None,
+        stride: int = 0,
+        truncetion_strategy: str = "longest_first",
     ) -> None:
         self._tokenizer = AutoTokenizer.from_pretrained(model_name)
-        default_start_tokens, default_end_tokens = _guess_start_and_end_token_defaults(model_name)
-        self._start_tokens = start_tokens if start_tokens is not None else default_start_tokens
-        self._end_tokens = end_tokens if end_tokens is not None else default_end_tokens
+        self._add_special_tokens = add_special_tokens
+        self._max_length = max_length
+        self._stride = stride
+        self._truncation_strategy = truncetion_strategy
 
-    @overrides
-    def tokenize(self, text: str) -> List[Token]:
+    def _tokenize(self, sentence_1: str, sentence_2: str = None):
+        """
+        This method works on both sentence and sentence pair.
+        """
         # TODO(mattg): track character offsets.  Might be too challenging to do it here, given that
         # ``transformers``` is dealing with the whitespace...
-        token_strings = self._start_tokens + self._tokenizer.tokenize(text) + self._end_tokens
-        return [Token(t) for t in token_strings]
 
+        encoded_tokens = self._tokenizer.encode_plus(
+            text=sentence_1,
+            text_pair=sentence_2,
+            add_special_tokens=self._add_special_tokens,
+            max_length=self._max_length,
+            stride=self._stride,
+            truncation_strategy=self._truncation_strategy,
+            return_tensors=None,
+        )
+        # token_ids containes a final list with ids for both regualr and special tokens
+        token_ids, token_type_ids = encoded_tokens["input_ids"], encoded_tokens["token_type_ids"]
+
+        tokens = []
+        for token_id, token_type_id in zip(token_ids, token_type_ids):
+            token_str = self._tokenizer.convert_ids_to_tokens(token_id, skip_special_tokens=False)
+            tokens.append(Token(text=token_str, text_id=token_id, type_id=token_type_id))
 
-def _guess_start_and_end_token_defaults(model_name: str) -> Tuple[List[str], List[str]]:
-    if "bert" in model_name:
-        return (["[CLS]"], ["[SEP]"])
-    else:
-        return ([], [])
+        return tokens
+
+    @overrides
+    def tokenize_sentences(self, sentences: List[str]) -> List[Token]:
```

Comment:
This seems to be too general, even as a method on this specific class - it's not possible to tokenise more than 2 sentences, so why allow it to be passed more than two sentences?

Reply:
I thought in the future we could implement automatic addition of special tokens in case of 3+ sentences. This might be useful for BERT based reading comprehension or summarization.